### PR TITLE
WP Media Library Picker

### PR DESCRIPTION
**Pull request by 0nko**
This is a continuation of the Media Picker refactoring project, which adds the new WordPress media library source module from the library. This PR removes the old media picker, hence the many line changes. 

The WP media picker needs to know about the site, so it is injected instead of being supplied during construction in order to keep the core library specific source-agnostic.

The new media picker is now used also for the Downloadable files.

image

### CODE REVIEW

**Review by anitaa1990 — APPROVED** [review]
Tested with WordPress media picker when adding product images and when adding downloadable files and everything works great 🎉 

I faced this one small issue but I'm good to merge this and take it up in another PR as well 👍  Nice work @0nko 🥇 
We might want to change the toolbar colors for the WordPress media picker screen to match the colors with our app. 
.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerLoaderFactory.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package com.woocommerce.android.mediapicker
+
+import org.wordpress.android.mediapicker.api.MediaPickerSetup
+import org.wordpress.android.mediapicker.loader.MediaLoader
+import org.wordpress.android.mediapicker.loader.MediaLoaderFactory
+import org.wordpress.android.mediapicker.source.device.DeviceMediaSource
+import org.wordpress.android.mediapicker.source.wordpress.MediaLibrarySource
+import javax.inject.Inject
+
+class MediaPickerLoaderFactory @Inject constructor(
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerLoaderFactory.kt:11]: np: A comment here on what this done would be beneficial.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerUtil.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import androidx.appcompat.app.AppCompatActivity
+import com.woocommerce.android.model.Product
+import com.woocommerce.android.util.WooLog
+import org.wordpress.android.mediapicker.MediaPickerConstants
+import org.wordpress.android.mediapicker.api.MediaPickerSetup
+import org.wordpress.android.mediapicker.model.MediaItem
+import org.wordpress.android.util.DateTimeUtils
+import java.security.InvalidParameterException
+
+object MediaPickerUtil {
```
Comment (anitaa1990) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/mediapicker/MediaPickerUtil.kt:16]: np: Similarly, a comment here on what this done would be beneficial.

**Review by mokagio — COMMENTED** [review]
I wouldn't know how to review the Kotlin code, but I verified that the app still builds when using composite build for the Media Picker library 👍 

Screen Shot 2021-11-18 at 3 06 11 pm

**Comment by 0nko** [comment]
Thanks for the review, @anitaa1990! 

> We might want to change the toolbar colors for the WordPress media picker screen to match the colors with our app.

@adamzelinski What color would you use here?

**Comment by ghost** [comment]
@0nko thanks for looking into this. It's looking great so far however, I think the designs at this stage look very different from that of what we have in iOS. I'll add some screenshots here but feel free to reach out if you have any other thoughts. CC @anitaa1990 

Screen Shot 2021-11-19 at 1 05 36 pm

> We might want to change the toolbar colors for the WordPress media picker screen to match the colors with our app.

> @adamzelinski What color would you use here?

As for the header can we please just use the current Android headers we use for light and dark modes. Thanks so much!
Screen Shot 2021-11-19 at 1 02 42 pm

**Comment by 0nko** [comment]
Thanks for the review, @adamzelinski! I've updated the colors to match the light/dark mode settings.

> I think the designs at this stage look very different from that of what we have in iOS

That's possible. I've been trying to keep the original Android look an feel. Plus that's what the WordPress app currently looks like.

Woo:

image

WordPress:

image

Would you like to also change the Done checkmark button to "ADD X"?

**Comment by ghost** [comment]
@0nko looking really great thanks! I still think we need to make the header look like our current Android ones (without the status colour & colored divider at the bottom) if that's okay thanks.

& yes please, let's use Add X for the confirmation.

**Comment by 0nko** [comment]
@adamzelinski Ready for another round. The slight off-white/black toolbar shade when an image is selected is in line with the WordPress app.

image

image

**Comment by ghost** [comment]
Looking really good now thanks for making those changes @0nko 

If we're able to remove or even just match the color of that pink bar at the bottom of the header to match the header colour then I think we're good. :)

Screen Shot 2021-11-30 at 4 10 29 pm

**Comment by 0nko** [comment]
@adamzelinski Sorry, I missed your comment about the divider earlier... It's fixed. I also managed to remove the tint so the toolbar background color is consistent now.

image

**Pull request merged**